File: solutions/cpp/pangram/1/pangram.cpp
#include "pangram.h"

using namespace std;
namespace pangram {
    bool is_pangram (const char* sentence){
        int lc[26] = {};

        if (strlen(sentence)==0) return false;
        for (size_t si = 0; si<strlen(sentence); si++){
            if (tolower(sentence[si])>='a' && 
                tolower(sentence[si])<='z') {
                lc[tolower(sentence[si])-'a']++;
            }
        }
        for (size_t lci = 0; lci<26; lci++){
            if (lc[lci]==0) {        
                return false;
            }
        }
        return true;
    }
}  // namespace pangram

File: solutions/fortran/pangram/1/pangram.f90
module pangram
  implicit none
contains

  logical function is_pangram(sentence)
    character(*) :: sentence
    character(26) :: lc,uc
    integer :: i
    lc="abcdefghijklmnopqrstuvwxyz"
    uc="ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    do i=1,len(sentence(:))
      if (index(lc,sentence(i:i))>0) then
        lc(index(lc,sentence(i:i)): &
              index(lc,sentence(i:i)))=" "
      end if
      if (index(uc,sentence(i:i))>0) then
        lc(index(uc,sentence(i:i)): &
              index(uc,sentence(i:i)))=" "
      end if
    end do
    if (len_trim(lc)==0) then
      is_pangram=.true.
      return
    end if
    is_pangram=.false.
    return  
   end function is_pangram

end module pangram
